### DISCUSSION TITLE
Only use `+page.server.svelte` on server side render

### DISCUSSION THREAD

**Discussion by jloh** [sveltejs/kit#8178] [Q&A] [answered]
Hi all. First off congrats on the v1 release!

I've recently migrated an app (https://github.com/jloh/geojs-app) over to v1 and am having trouble migrating after the removal of `$session` (https://github.com/sveltejs/kit/discussions/5883).

For my app to correctly render server side (which I'd like!) I need to get the end users IP address. Previously I could use `session.userIP` but I now need to use the request event and use `event.getClientAddress()`. This works when its done in `+page.server.js` and ends up looking like this:

```js
/** @type {import('./$types').PageServerLoad} */
export async function load(event) {
	return {
		clientIP: event.getClientAddress()
	};
}
```

Then my `+page.js` looks like this:

```js
/** @type {import('./$types').PageLoad} */
import { browser } from '$app/environment';
export const prerender = false;
export async function load({ fetch, data }) {
	const { clientIP } = data;
	var geo, ptr;
	if (browser) {
		geo = 'https://get.geojs.io/v1/ip/geo.json';
		ptr = 'https://get.geojs.io/v1/dns/ptr.json';
	} else {
		geo = `https://get.geojs.io/v1/ip/geo/${clientIP}.json`;
		ptr = `https://get.geojs.io/v1/dns/ptr/${clientIP}.json`;
	}
	const geoResponse = await fetch(geo);
	const ptrResponse = await fetch(ptr);
	return {
		geoLookup: geoResponse.ok && (await geoResponse.json()),
		ptrLookup: ptrResponse.ok && ptrResponse.json()
	};
}
```

This works and when running server side I correctly request the requesting users IP.

The problem is if someone navigates away then back to the homepage `+page.server.svelte` executes again when it isn't required to, after initial render the page can run completely client side.

What I'm wondering is can I get the clients address in a normal `+page.js` still or is it only available in `+page.server.js`? And alternatively can I make the `+page.server.js` only execute when running server side otherwise let the app run completely client side?

Hopefully this makes sense and thanks in advance!

**Comment by 1MuhammadYahya** [comment, marked as answer]
I think the problem is that the page load function runs everytime the page is loaded. Regardless of CSR or SSR. The load function would always be called. Why dont you instead send these requests from the server and return the responses to the client to be handled or just use a hook?

**Reply by jloh** [reply]
Thankyou! Using a server hook is really what I should’ve been doing and worked perfectly for me.

**Comment by Jakeii** [comment]
I think this is similar to a request I made a while ago https://github.com/sveltejs/kit/discussions/6566, I think there's definitely a desire and use case for this, lower level control over how the load functions work.
